src/software/jetson_nano/services/motor.cpp

```diff
@@ -624,6 +628,40 @@ void MotorService::spiTransfer(int fd, uint8_t const* tx, uint8_t const* rx, uns
 }
 
 
+void MotorService::readThenWriteSpiTransfer(int fd, const uint8_t* read_tx,
+                                            const uint8_t* write_tx,
+                                            const uint8_t* read_rx, uint32_t spi_speed)
+{
+    int ret1, ret2;
+
+    uint8_t write_rx[5] = {0};
+
+    struct spi_ioc_transfer tr[2];
+    memset(tr, 0, sizeof(tr));
+
+    tr[0].tx_buf        = (unsigned long)read_tx;
+    tr[0].rx_buf        = (unsigned long)read_rx;
+    tr[0].len           = 5;
+    tr[0].delay_usecs   = 0;
+    tr[0].speed_hz      = spi_speed;
+    tr[0].bits_per_word = 8;
+    tr[0].cs_change     = 0;
+    tr[1].tx_buf        = (unsigned long)write_tx;
+    tr[1].rx_buf        = (unsigned long)write_rx;
+    tr[1].len           = 5;
+    tr[1].delay_usecs   = 0;
+    tr[1].speed_hz      = spi_speed;
+    tr[1].bits_per_word = 8;
+    tr[1].cs_change     = 0;
+
+    ret1 = ioctl(fd, SPI_IOC_MESSAGE(1), &tr);
+    ret2 = ioctl(fd, SPI_IOC_MESSAGE(1), &tr[1]);
+
+    CHECK(ret1 >= 1 && ret2 >= 1)
+        << "SPI Transfer to motor failed, not safe to proceed: errno " << strerror(errno);
```

Comment:
I'm not familiar with SPI, but if these invariants were false, does that mean that future SPI transfers will be messed up? Or does it just mean that this iteration did not go though as expected. 

Since if it's just this iteration that's effected, then I don't think having a `CHECK` is necessarily a good idea as it will crash the program.

Reply:
I'm not sure, we have this CHECK in the other spi transfer code. If there's noise on the spi line that prevents us from safely transferring data, then maybe it's safe to crash

we still don't know why motor runaway happens and it's probably safer to crash. what do you think?